### ISSUE TITLE
wrong APK at tag 2.3.4?

### ISSUE THREAD

**Issue by IzzySoft** [achunt2143/W-Launcher#13]
The APK at release v2.3.4 identifies as

    package: name='com.achunt.weboslauncher' versionCode='32' versionName='2.3.3'

Did you attach the wrong APK maybe?

**Comment by IzzySoft** [comment]
@achunt2143 any chance to get that fixed? Our updater fetches the APK on each run, just to throw it away again. Should you no longer maintain this project, we can of course disable updates altogether, just let us know then please.

**Comment by achunt2143** [comment]
Oh my bad. I didn't even notice this. Updates have been slow due to personal things but next few days I'll try to get this corrected.

**Comment by IzzySoft** [comment]
Thanks! Be welcome to give me a ping here once a new release is up, so I can check and adjust accordingly (until then, update checks for W-Launcher are demoted from daily to monthly).

**Comment by achunt2143** [comment]
I have corrected this. Can you please check it?

**Comment by IzzySoft** [comment]
Seems to be fine:

```
$ iod repo get com.achunt.weboslauncher -v
com.achunt.weboslauncher: looking for 'https://api.github.com/repos/achunt2143/W-Launcher/releases'
com.achunt.weboslauncher: checking tag 'release-apk-2.3.5'
com.achunt.weboslauncher: lastRelNo set to 'release-apk-2.3.5', checking for files
com.achunt.weboslauncher: Upstream file date (2025-06-05 17:46) is newer than ours (2024-07-25 20:29).
com.achunt.weboslauncher: returning ['release-apk-2.3.5','https://github.com/achunt2143/W-Launcher/releases/download/release-apk-2.3.5/app-release2.3.5.apk',1749138390]
com.achunt.weboslauncher: 2.3.3/release-apk-2.3.5, https://github.com/achunt2143/W-Launcher/releases: https://github.com/achunt2143/W-Launcher/releases/download/release-apk-2.3.5/app-release2.3.5.apk
- Grabbing update for com.achunt.weboslauncher: OK
- Checking 'repo/com.achunt.weboslauncher_34.apk' for libraries and malware …
- Checking the app's AndroidManifest.xml …
! repo/com.achunt.weboslauncher_34.apk declares sensitive permission(s): android.permission.QUERY_ALL_PACKAGES
! repo/com.achunt.weboslauncher_34.apk contains signature block blobs: 0x504b4453 (DEPENDENCY_INFO_BLOCK; GOOGLE)
- Cloning upstream repo …
- Checking Fastlane for updates …
com.achunt.weboslauncher: check if repo contains FUNDING.yml
com.achunt.weboslauncher: looking for 'https://api.github.com/repos/achunt2143/.github/contents/'
com.achunt.weboslauncher: Github reports "Not Found" for https://api.github.com/repos/achunt2143/.github/contents/
com.achunt.weboslauncher: no FUNDING.yml detected.
com.achunt.weboslauncher: calling 'getFastlaneMetaLocal('local',[host:localhost,owner:achunt2143,repo:W-Launcher,path:fastlane/metadata/android],0
com.achunt.weboslauncher: FastlaneFeatures shortdesc,fulldesc,icon,screenshotsJPG
com.achunt.weboslauncher: could not find local fastlane; '/web/ftp/repo/fdroid/build/com.achunt.weboslauncher/repo/fastlane/metadata/android' does either not exist, is no directory, or not accessible.
```

What the log also shows, are some open questions: what is `QUERY_ALL_PACKAGES` used/needed for? As for the `DEPENDENCY_INFO_BLOCK`, that can easily be avoided by adding some lines to your `build.gradle` for your future releases:

```gradle
android {
    dependenciesInfo {
        // Disables dependency metadata when building APKs (for IzzyOnDroid/F-Droid)
        includeInApk = false
        // Disables dependency metadata when building Android App Bundles (for Google Play)
        includeInBundle = false
    }
}
```

For some background: that BLOB is supposed to be just a binary representation of your app's dependency tree. But as it's encrypted with a public key belonging to Google, only Google can read it – and nobody else can even verify what it really contains. More details can be found e.g. here: [Ramping up security: additional APK checks are in place with the IzzyOnDroid repo](https://android.izzysoft.de/articles/named/iod-scan-apkchecks#blobs).

**Comment by achunt2143** [comment]
Thanks for all that info. I didn't know google added this kind of stuff. QUERY_ALL_PACKAGES is the permission needed to get all the applications installed on the device. I'll try one more release with that dependenciesInfo flag and ping you here once it's up. If it's not a critical addition, I've been working on some new features and general enhancements last few months. I'll see if I can wrap those up.

**Comment by IzzySoft** [comment]
> I didn't know google added this kind of stuff.

That Blob was introduced with some AGP update a bit more than a year ago, without "big announcement" (just mentioned somewhere in the "small print") – so most missed it first, us included. Came up with our signing block scanner, so we wondered what caused it – which was how we found out…

> I'll try one more release with that dependenciesInfo flag

With – or without? It's only useful for Playstore, and garbage for all else. If you upload the AAB to play, you could leave the blob in for that (`includeInBundle`) – but we'd appreciate having it gone from the APK.

> If it's not a critical addition

Well, "next release" didn't state any **time** frame, did it? 😉 Whenever the next release is ready. Adding those lines should have no side-effects.

> QUERY_ALL_PACKAGES is the permission needed to get all the applications installed on the device

Eh, stupid me: this is a **launcher** 🙈 Yeah, no more questions on that, your honor 🤣 Adding it straight to the app's "green list".

**Comment by achunt2143** [comment]
@IzzySoft you're right I didn't say when ;) until now. I just release new one with the dependenciesInfo flag set false, and included at least one new feature :)

**Comment by IzzySoft** [comment]
```
$ iod repo get com.achunt.weboslauncher -v
com.achunt.weboslauncher: looking for 'https://api.github.com/repos/achunt2143/W-Launcher/releases'
com.achunt.weboslauncher: checking tag 'release-apk-2.3.6'
com.achunt.weboslauncher: lastRelNo set to 'release-apk-2.3.6', checking for files
com.achunt.weboslauncher: Upstream file date (2025-06-13 21:12) is newer than ours (2025-06-05 19:35).
com.achunt.weboslauncher: returning ['release-apk-2.3.6','https://github.com/achunt2143/W-Launcher/releases/download/release-apk-2.3.6/app-release2.3.6.apk',1749841969]
com.achunt.weboslauncher: 2.3.5/release-apk-2.3.6, https://github.com/achunt2143/W-Launcher/releases: https://github.com/achunt2143/W-Launcher/releases/download/release-apk-2.3.6/app-release2.3.6.apk
- Grabbing update for com.achunt.weboslauncher: OK
- Checking 'repo/com.achunt.weboslauncher_35.apk' for libraries and malware …
- Checking the app's AndroidManifest.xml …
- Cloning upstream repo …
! com.achunt.weboslauncher: last commit (from tag 'release-apk-2.3.6') is too old (27912054s/324d), cloning from HEAD instead
- Checking Fastlane for updates …
com.achunt.weboslauncher: check if repo contains FUNDING.yml
com.achunt.weboslauncher: replacing 'build/com.achunt.weboslauncher/FUNDING.yml'
com.achunt.weboslauncher: calling 'getFastlaneMetaLocal('local',[host:localhost,owner:achunt2143,repo:W-Launcher,path:fastlane/metadata/android],0
com.achunt.weboslauncher: FastlaneFeatures shortdesc,fulldesc,icon,screenshotsJPG
com.achunt.weboslauncher: parsing local fastlane in '/web/ftp/repo/fdroid/build/com.achunt.weboslauncher/repo/fastlane/metadata/android'
com.achunt.weboslauncher: checking locale 'en-US'
com.achunt.weboslauncher: updating 'metadata/com.achunt.weboslauncher/en-US/full_description.txt'
com.achunt.weboslauncher: updating 'metadata/com.achunt.weboslauncher/en-US/short_description.txt'
com.achunt.weboslauncher: updating 'repo/com.achunt.weboslauncher/en-US/icon.png'
com.achunt.weboslauncher: updating 'repo/com.achunt.weboslauncher/en-US/phoneScreenshots/01-Screenshot_home.jpg' as JPG
com.achunt.weboslauncher: updating 'repo/com.achunt.weboslauncher/en-US/phoneScreenshots/02-Screenshot_appdrawer.jpg' as JPG
com.achunt.weboslauncher: updating 'repo/com.achunt.weboslauncher/en-US/phoneScreenshots/03-Screenshot_settings.jpg' as JPG
com.achunt.weboslauncher: cross-checking for obsolete screenshots
com.achunt.weboslauncher: screenshots in Fastlane: 01-Screenshot_home,02-Screenshot_appdrawer,03-Screenshot_settings
com.achunt.weboslauncher: local screenshots checked: 01-Screenshot_home,02-Screenshot_appdrawer,03-Screenshot_settings
```

What confuses me is the `last commit (from tag 'release-apk-2.3.6') is too old (27912054s/324d)`. You're sure you tagged the correct commit?

![Image](https://github.com/user-attachments/assets/ece9d632-3f03-4da2-8354-4228cdc09156)

**Comment by achunt2143** [comment]
You're right. I didn't commit my change before hitting submit on the release. I updated it now.

**Comment by IzzySoft** [comment]
Did the APK change, or is it the same file as before? It was not yet published here, so if its a different file, now would be the time to replace it here. Once it's out it's too late.

**Comment by achunt2143** [comment]
No the file itself is correct. It was built with the correct code.

**Comment by IzzySoft** [comment]
The APK seems to have been built from c81f886724d162dd12d0f7dfa93494e1cc932f8c – but the versioning change was pushed with the next commit. I hope there were no changes between building and committing at least 😉 Let's see… Hm, close but no cigar; guess you're no longer on JDK-17 but use JDK-21, if my nose is guiding me right there (going by the differences in the DEX file compared to our build on JDK-17)… Yes, it did! OK, so…

Congrats, your APK is reproducible! 🥳 Why didn't I test that before? 🤔 Nevermind, did it now. So this release will have its green shield up by tomorrow (with the next sync around 6 pm UTC).

Do you like badges, and shields? Be welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to https://apt.izzysoft.de/packages/com.achunt.weboslauncher right away, and an RB shield as soon as the sync has completed 😃 `[<img src="https://shields.rbtlog.dev/simple/com.achunt.weboslauncher" alt="RB shield">](https://shields.rbtlog.devcom.achunt.weboslauncher)` should become available within the next 6 h.

**Comment by achunt2143** [comment]
I think you're right. I think i might have java 21 on my laptop instead of java 17 like it is on my desktop. I'll pick a badge! I really like the original without border. I'll add it with a link to your repo. And I'll try and get some screenshot updated too!

**Comment by IzzySoft** [comment]
Fastlane updates would be automatically pulled with the next release then:

```yaml
  Fastlane: https://github.com/achunt2143/W-Launcher/tree/main/fastlane/metadata/android
  FastlaneFeatures: shortdesc,fulldesc,icon,screenshotsJPG
```

(`screenshotsJPG` means screenshots would be converted to JPG; that was done for size optimization. If you give me a ping before they are pulled, I can check if that's still needed. And should you add per-release changelogs, let me know too, please, so I enable them as well)

And yes, pick the badge you feel best fitting. That's why we have a selection to choose from 😉 There will be more RB shields one day, too, I've heard…

**Comment by achunt2143** [comment]
This one too pls 😊

**Comment by IzzySoft** [comment]
per-release changelogs added, and using `screenshots` instead of `screenshotsJPG` now. Do I see it correctly that we should focus on tags starting with `release-apk-` now, and ignore all others?

**Comment by achunt2143** [comment]
yes correct. tags that have a proper release will be marked `release-apk-` and the version number. others, if they release would be marked with testing-apk

**Comment by IzzySoft** [comment]
Thanks! Adjusted our configs to only pick tags starting with `release-apk-` then.
